robocop/checkers/naming.py

```diff
@@ -58,48 +61,197 @@ def visit_KeywordName(self, node):  # noqa
 
-    def visit_Setup(self, node):
-        self.check_if_keyword_is_capitalized(node.name, node)
+    def visit_Setup(self, node):  # noqa
+        self.check_keyword_naming(node.name, node)
 
-    def visit_SuiteTeardown(self, node):
-        self.check_if_keyword_is_capitalized(node.name, node)
+    def visit_SuiteTeardown(self, node):  # noqa
+        self.check_keyword_naming(node.name, node)
 
-    def visit_TestTeardown(self, node):
-        self.check_if_keyword_is_capitalized(node.name, node)
+    def visit_TestTeardown(self, node):  # noqa
+        self.check_keyword_naming(node.name, node)
 
-    def visit_Teardown(self, node):
-        self.check_if_keyword_is_capitalized(node.name, node)
+    def visit_Teardown(self, node):  # noqa
+        self.check_keyword_naming(node.name, node)
 
     def visit_TestCase(self, node):  # noqa
-        self.generic_visit(node)
+        if self.is_comment(node.name, node):
+            return
+        else:
+            self.generic_visit(node)
 
     def visit_Keyword(self, node):  # noqa
-        if not node.name.lstrip().startswith('#'):
-            self.check_if_keyword_is_capitalized(node.name, node)
+        self.check_keyword_naming(node.name, node)
         self.generic_visit(node)
 
     def visit_KeywordCall(self, node):  # noqa
-        self.check_if_keyword_is_capitalized(node.keyword, node)
+        self.check_keyword_naming(node.keyword, node)
 
-    def check_if_keyword_is_capitalized(self, keyword_name, node):  # noqa
+    def is_comment(self, name, node):
+        if name.lstrip().startswith('#'):
+            self.report("invalid-comment", node=node)
+            return True
+        return False
+
+    def check_keyword_naming(self, keyword_name, node):  # noqa
+        if self.is_comment(keyword_name, node):
+            return
         if keyword_name == r'/':  # old for loop, / are interpreted as keywords
             return
+        if keyword_name.startswith('...'):
+            self.report("not-enough-whitespace-after-newline-marker", node=node)
+            return
+        if normalize_robot_name(keyword_name) == 'runkeywordif':
+            for token in node.data_tokens:
+                if (token.value.lower() in self.else_if) and not token.value.isupper():
+                    self.report(
+                        "keyword-name-is-reserved-word",
+                        token.value,
+                        self.prepare_reserved_word_rule_message(token.value, 'Run Keyword If'),
+                        node=node
+                    )
+        elif self.check_if_keyword_is_reserved(keyword_name, node):
+            return
         words = keyword_name.replace('_', ' ').split(' ')
-        if any(not word.istitle() for word in words):
+        if any(not (word.istitle() or word.isupper()) for word in words if not self.is_variable(word)):
             self.report("not-capitalized-keyword-name", node=node)
+
+    def is_variable(self, name):
+        if len(name) < 3:
```

Comment:
I think it can be even `< 4`

Reply:
You're right, ${a} is already 4, and ${} is not variable (probably, I will test ;))